client/api/src/call_executor.rs

```diff
@@ -29,6 +29,16 @@ use sp_externalities::Extensions;
 use sp_core::NativeOrEncoded;
 
 use sp_api::{ProofRecorder, InitializeBlock, StorageTransactionCache};
+use crate::execution_extensions::ExecutionExtensions;
+
+/// Executor Provider
+pub trait ExecutorProvider<Block: BlockT, E: CallExecutor<Block>> {
+	/// Get call executor reference.
+	fn executor(&self) -> &E;
```

Comment:
Having this makes the bounds more complicated in so many places, in what context is it being used?
Maybe it's worth to consider returning a `dyn` here or making `CallExecutor` and associated type? Basically I'm curious if less code complication is worth the dynamic call - maybe we just use it once in RPC or sth.

Reply:
fixed this.